db/commitlog/commitlog.cc

```diff
@@ -205,8 +205,19 @@ class db::commitlog::segment_manager : public ::enable_shared_from_this<segment_
     bool _shutdown = false;
     std::optional<shared_promise<>> _shutdown_promise = {};
 
+    struct request_controller_timeout_exception_factory {
+        class request_controller_timed_out_error : public timed_out_error {
+        public:
+            virtual const char* what() const noexcept override {
+                return "request_controller: timed out";
```

Comment:
I think this message needs some more context, `request_controller` is a too generic name.

Reply:
Having no better idea, I wrote `segment_manager::_request_controller`